src/boot/index.js

```diff
@@ -9,18 +9,59 @@
 
 /* global __MANIFEST__ */
 
+/**
+ * Subset of configuration settings (see https://h.readthedocs.io/projects/client/en/latest/publishers/config/)
+ * which affect where the client loads its assets from.
+ *
+ * @typedef BootSettings
+ * @prop {string} [assetRoot]
+ * @prop {string} [sidebarAppUrl]
+ */
+
 import { jsonConfigsFrom } from '../shared/settings';
 
 import boot from './boot';
 import processUrlTemplate from './url-template';
 
+/** @type {BootSettings} */
 const settings = jsonConfigsFrom(document);
 
+// Use the asset root and sidebar app locations specified in the host page,
+// if present. This is used by the Hypothesis browser extensions to make the
+// boot script load assets bundled with the extension.
+let assetRoot;
+if (settings.assetRoot) {
+  // The `assetRoot` setting is assumed to point at the root of the contents of
+  // the npm package.
+  assetRoot = settings.assetRoot + 'build/';
+}
+let sidebarAppUrl = settings.sidebarAppUrl;
+
+// Otherwise, try to determine the default root URL for assets and the sidebar
+// application from the location where the boot script is hosted.
+try {
+  const script = /** @type {HTMLScriptElement} */ (document.currentScript);
+  let scriptUrl = new URL(script.src);
+
+  // We only use the bundled sidebar HTML and assets if the boot script has
```

Comment:
I don't propose a better solution, but this does feel fragile and hard to understand.  I wonder if there is something that can be driven from the config itself to trigger this condition to make it feel a bit more explicit.